### PR TITLE
Upgraded 'Adventure' samples

### PR DESCRIPTION
**Pull request by IEvangelist**
Contributes to https://github.com/dotnet/orleans/issues/8035

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/orleans/pull/8120)

### CODE REVIEW

File: samples/Adventure/AdventureClient/Program.cs [hunk 1]
```diff
@@ -1,24 +1,29 @@
...
+await host.StartAsync();
 
-Console.WriteLine(@"
-  ___      _                 _                  
- / _ \    | |               | |                 
-/ /_\ \ __| |_   _____ _ __ | |_ _   _ _ __ ___ 
-|  _  |/ _` \ \ / / _ \ '_ \| __| | | | '__/ _ \
-| | | | (_| |\ V /  __/ | | | |_| |_| | | |  __/
-\_| |_/\__,_| \_/ \___|_| |_|\__|\__,_|_|  \___|");
+Console.WriteLine("""
```
Comment (ReubenBond) [inline samples/Adventure/AdventureClient/Program.cs:12]: 😎

File: samples/Adventure/AdventureGrainInterfaces/Thing.cs [hunk 2]
```diff
@@ -1,32 +1,30 @@
...
-    long Id,
-    string Name,
-    string Category,
-    long FoundIn,
-    List<string> Commands);
+    [property: Id(0)] long Id,
+    [property: Id(1)] string Name,
+    [property: Id(2)] string Category,
+    [property: Id(3)] long FoundIn,
+    [property: Id(4)] List<string> Commands);
```
Comment (ReubenBond) [inline samples/Adventure/AdventureGrainInterfaces/Thing.cs:9]: Id attributes aren't necessary on properties in the primary constructor by default

**Review by ReubenBond — APPROVED** [review]
Minor note about ids not being necessary in primary constructors

File: samples/Adventure/AdventureGrainInterfaces/Thing.cs (hunk 2 above)
Comment (IEvangelist) [reply samples/Adventure/AdventureGrainInterfaces/Thing.cs]: removed

**Pull request merged**